src/Ray/Math.hs

```diff
@@ -64,7 +64,13 @@ intersect start ray (OCircle (Circle plane radius) color) = do
   let r1 = distance (origin plane) (start + fmap (* len) ray)
   guard (r1 <= radius)
   pure i
-intersect start ray (OSquare Square {..} color) = Nothing
+intersect start ray (OSquare Square {..} color) = do
+  i@(Intersection _ len) <- intersectPlane start ray color plane
+  let inPlane = (start + fmap (* len) ray) - origin plane
```

Comment:
Though I'm not sure `fmap (* len)` is the right way to scale a vector by a constant...

Reply:
No, apparently it's the right way there's even an operator for that: https://www.stackage.org/haddock/lts-15.15/linear-1.20.9/src/Linear.Vector.html#%2A%5E
:smiley: